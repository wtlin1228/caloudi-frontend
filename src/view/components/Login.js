import React from 'react'
import styled from 'styled-components'
import { Input } from './Inputs'
import Form from './Form'

const StyledLabel = styled.label`
  margin-right: 10px;
`

function Login({ onSubmit }) {
  return (
    <div>
      <Form
        onSubmit={e => {
          e.preventDefault()
          const { username, password } = e.target.elements
          onSubmit({
            username: username.value,
            password: password.value
          })
        }}
      >
        <StyledLabel htmlFor="username-input">Username</StyledLabel>
        <Input
          id="username-input"
          placeholder="Username..."
          name="username"
          style={{ flex: 1, marginBottom: 20 }}
        />
        <StyledLabel id="password-input">Password</StyledLabel>
        <Input
          placeholder="Password..."
          type="password"
          name="password"
          aria-labelledby="password-input"
        />
      </Form>
    </div>
  )
}

export default Login
