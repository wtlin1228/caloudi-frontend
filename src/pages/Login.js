import React, { useEffect } from 'react'
import PropTypes from 'prop-types'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'

// utils

// assets

// actions
import { authAsyncActions } from 'Auth'

// components
import { LoginForm } from 'components'

// self-defined-components

const Login = ({ isLogin, loginAsync }) => {
  useEffect(() => {
    if (isLogin) {
      window.location.reload()
    }
  }, [isLogin])

  function handleSubmit({ username, password }) {
    loginAsync({
      userId: username,
      password
    })
  }

  return (
    <div>
      <LoginForm onSubmit={handleSubmit} />
    </div>
  )
}

Login.propTypes = {
  isLogin: PropTypes.bool.isRequired,
  loginAsync: PropTypes.func.isRequired
}

const mapStateToProps = ({ auth }) => ({
  isLogin: auth.isLogin
})

const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      loginAsync: authAsyncActions.loginAsync
    },
    dispatch
  )

export default connect(mapStateToProps, mapDispatchToProps)(Login)
