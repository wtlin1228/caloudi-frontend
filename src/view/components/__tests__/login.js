import React from 'react'
import { render, cleanup, Simulate } from '@testing-library/react'
import Login from '../Login'

afterEach(cleanup)

test('calls onSubmit with the username and password when submitted', () => {
  // Arrange
  const fakeUser = {
    username: 'sandbox.dev@caloudi.com',
    password: 'cspdev4fun#'
  }
  const handleSubmit = jest.fn()
  const { container, getByLabelText, getByText } = render(
    <Login onSubmit={handleSubmit} />
  )

  const usernameNode = getByLabelText('Username')
  const passwordNode = getByLabelText('Password')
  const formNode = container.querySelector('form')
  const submitButtonNode = getByText('Submit')

  usernameNode.value = fakeUser.username
  passwordNode.value = fakeUser.password

  // Act
  Simulate.submit(formNode)

  // Assert
  // no change necessary here
  expect(handleSubmit).toHaveBeenCalledTimes(1)
  expect(handleSubmit).toHaveBeenCalledWith(fakeUser)
  expect(submitButtonNode.type).toBe('submit')
})

test('snapshot', () => {
  const { container } = render(<Login />)
  expect(container.firstChild).toMatchSnapshot()
})
