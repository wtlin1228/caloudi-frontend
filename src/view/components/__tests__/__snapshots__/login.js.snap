// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div>
  <div
    class="sc-AxgMl iKWKjR"
  >
    <form
      class="sc-AxhCb dbVdvl"
    >
      <div
        class="sc-AxhUy lfhkHP"
      >
        <label
          class="sc-AxheI zyOpI"
          for="username-input"
        >
          Username
        </label>
        <input
          class="sc-AxjAm dSgLgW"
          id="username-input"
          name="username"
          placeholder="Username..."
          style="flex: 1; margin-bottom: 20px;"
        />
        <label
          class="sc-AxheI zyOpI"
          id="password-input"
        >
          Password
        </label>
        <input
          aria-labelledby="password-input"
          class="sc-AxjAm dSgLgW"
          name="password"
          placeholder="Password..."
          type="password"
        />
      </div>
      <button
        class="sc-AxiKw fdWJBE"
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</div>
`;
